burr/core/application.py

```diff
@@ -730,7 +751,7 @@ def streaming_response(state: State, prompt: str) -> Generator[dict, None, Tuple
             raise ValueError(
                 f"Cannot stream result -- no next action found! Prior action was: {self._state[PRIOR_STEP]}"
             )
-
+        self._increment_sequence_id()  # TODO: is this in the right place?
```

Comment:
So this is the right place if we move incrementing sequence ID to before (this is a makeshift step) -- this does simplify things.